receiver/k8sclusterreceiver/internal/clusterresourcequota/clusterresourcequotas.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package clusterresourcequota // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/clusterresourcequota"
+
+import (
+	"strings"
+	"time"
+
+	quotav1 "github.com/openshift/api/quota/v1"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+
+	imetadataphase "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/clusterresourcequota/internal/metadata"
+)
+
+func GetMetrics(set receiver.CreateSettings, crq *quotav1.ClusterResourceQuota) pmetric.Metrics {
+	mbphase := imetadataphase.NewMetricsBuilder(imetadataphase.DefaultMetricsBuilderConfig(), set)
+	ts := pcommon.NewTimestampFromTime(time.Now())
+
+	for k, v := range crq.Status.Total.Hard {
+		val := v.Value()
+		if strings.HasSuffix(string(k), ".cpu") {
+			val = v.MilliValue()
+		}
+
+		mbphase.RecordOpenshiftClusterquotaLimitDataPoint(ts, val, string(k))
+	}
+
+	for k, v := range crq.Status.Total.Used {
+		val := v.Value()
+		if strings.HasSuffix(string(k), ".cpu") {
+			val = v.MilliValue()
+		}
+
+		mbphase.RecordOpenshiftClusterquotaUsedDataPoint(ts, val, string(k))
+	}
+
+	for _, ns := range crq.Status.Namespaces {
+		for k, v := range ns.Status.Hard {
+			val := v.Value()
+			if strings.HasSuffix(string(k), ".cpu") {
+				val = v.MilliValue()
+			}
+
+			mbphase.RecordOpenshiftAppliedclusterquotaLimitDataPoint(ts, val, ns.Namespace, string(k))
+		}
+
+		for k, v := range ns.Status.Used {
+			val := v.Value()
+			if strings.HasSuffix(string(k), ".cpu") {
+				val = v.MilliValue()
+			}
```

Comment:
It's cleaner now. You can extract this part to a function `extractValue(k, v)`

Reply:
Done.